hw4-ftp-gui: init commit

AntonYermilov opened the pull request:
Upgraded client-server interaction
Added gui
Updated tests

AntonYermilov committed:
hw4-ftp-gui: init commit
Upgraded client-server interaction
Added gui
Updated tests

jzuken commented on term2/hw4-ftp-gui/src/main/java/me/eranik/gui/MainActivity.java:
```diff
@@ -0,0 +1,53 @@
+    /**
+     * Starts app. Connects to server and lists files in the root directory.
+     * @param primaryStage specified primary stage.
+     * @throws Exception if any error occurred
```
а метод IOException бросает

AntonYermilov committed:
hw4-ftp-gui:
Updated tests
Fixed minor misprints
Ugraded GUI: now default place to store files is substituted to the text field

jzuken commented:
я в прошлый раз проглядел: эту ветку надо отводить от предыдущей задачи. вы же сюда просто код скопировали, а фиксы тоже руками переносить будете?

AntonYermilov commented:
@jzuken 
Неправда, что это просто скопированный код, он имеет ряд значительных отличий и, в частности, здесь немного иная архитектура и отсутствует большая часть тех проблем, которые есть в аналогичной задаче из другой ветки.

AntonYermilov committed:
hw4-ftp-gui:
Updated argument parsing
Added enums for client-server interaction

AntonYermilov committed:
hw4-ftp-gui
Updated query types in server implementation

jzuken commented:
так это здорово, надо эти улучшения в прошлой задаче тоже применить

AntonYermilov commented:
@jzuken
В другой задаче совсем иной интерфейс взаимодействия с пользователем. Немного по-иному устроена та же команда list и, как следствие, тесты. Ну и обработка ошибок отчасти отличается.

Те улучшения, которые очевидным образом применялись к прошлой задаче, уже и так были применены :)

jzuken commented:
если у вас детали реализации меняются из-за пользовательского интерфейса, хреново вы изначально эту реализацию продумали. здесь одна и та же функциональность, и я не вижу, почему их реализация должна различаться

jzuken commented:
всё ещё хочется gui в отдельной ветке пуллреквестом к задаче 3
